aptos-move/move-examples/token_objects/knight/sources/knight.move

```diff
@@ -0,0 +1,541 @@
+
+        // Initialize the property map and the knight condition as Hungry
+        let properties = property_map::prepare_input(vector[], vector[], vector[]);
+        property_map::init(&constructor_ref, properties);
+        property_map::add_typed(
+            &property_mutator_ref,
+            string::utf8(b"Condition"),
+            string::utf8(CONDITION_HUNGRY),
+        );
+
+        property_map::add_typed(
+            &property_mutator_ref,
+            string::utf8(b"Health Point"),
+            1,
+        );
+
+        // Publishes the knightToken resource with the refs and the event handle for `LevelUpdateEvent`.
+        let knight_token = KnightToken {
+            mutator_ref,
+            burn_ref,
+            property_mutator_ref,
+            health_update_events: object::new_event_handle(&object_signer),
+            base_uri
+        };
+        move_to(&object_signer, knight_token);
+    }
+
+    /// Mints the given amount of the corn token to the given receiver.
+    public entry fun mint_corn(creator: &signer, receiver: address, amount: u64) acquires FoodToken {
+        let corn_token = object::address_to_object<FoodToken>(corn_token_address());
+        mint_internal(creator, corn_token, receiver, amount);
+    }
+
+    /// Mints the given amount of the meat token to the given receiver.
+    public entry fun mint_meat(creator: &signer, receiver: address, amount: u64) acquires FoodToken {
+        let meat_token = object::address_to_object<FoodToken>(meat_token_address());
+        mint_internal(creator, meat_token, receiver, amount);
+    }
+
+    /// The internal mint function.
+    fun mint_internal(creator: &signer, token: Object<FoodToken>, receiver: address, amount: u64) acquires FoodToken {
+        let food_token = authorized_borrow<FoodToken>(creator, &token);
+        let fungible_asset_mint_ref = &food_token.fungible_asset_mint_ref;
+        let fa = fungible_asset::mint(fungible_asset_mint_ref, amount);
+        primary_fungible_store::deposit(receiver, fa);
+    }
+
+    /// Transfers the given amount of the corn token from the given sender to the given receiver.
+    public entry fun transfer_corn(from: &signer, to: address, amount: u64) {
+        transfer_food(from, object::address_to_object<FoodToken>(corn_token_address()), to, amount);
+    }
+
+    /// Transfers the given amount of the meat token from the given sender to the given receiver.
+    public entry fun transfer_meat(from: &signer, to: address, amount: u64) {
+        transfer_food(from, object::address_to_object<FoodToken>(meat_token_address()), to, amount);
+    }
+
+    public entry fun transfer_food(from: &signer, food: Object<FoodToken>, to: address, amount: u64) {
+        let metadata = object::convert<FoodToken, Metadata>(food);
+        primary_fungible_store::transfer(from, metadata, to, amount);
+    }
+
+    public entry fun burn_food(from: &signer, food: Object<FoodToken>, amount: u64) acquires FoodToken {
```

Comment:
It's rather strange to me that this is public entry as this would allow users to randomly burn food tokens (but not feeding them to knights). Should this be private instead?

Reply:
Done.